libs/ktem/ktem/index/file/pipelines.py

```diff
@@ -67,58 +67,64 @@ class DocumentRetrievalPipeline(BaseFileIndexRetriever):
             documents
         get_extra_table: if True, for each retrieved document, the pipeline will look
             for surrounding tables (e.g. within the page)
+        top_k: number of documents to retrieve
+        mmr: whether to use mmr to re-rank the documents
+        disabled: whether the pipeline is disabled
     """
 
     vector_retrieval: VectorRetrieval = VectorRetrieval.withx()
     reranker: BaseReranking
     get_extra_table: bool = False
+    mmr: bool = False
+    top_k: int = 5
 
     def run(
         self,
         text: str,
-        top_k: int = 5,
-        mmr: bool = False,
         doc_ids: Optional[list[str]] = None,
+        *args,
+        **kwargs,
     ) -> list[RetrievedDocument]:
         """Retrieve document excerpts similar to the text
 
         Args:
             text: the text to retrieve similar documents
-            top_k: number of documents to retrieve
-            mmr: whether to use mmr to re-rank the documents
             doc_ids: list of document ids to constraint the retrieval
         """
+        if not doc_ids:
+            logger.info(f"Skip retrieval because of no selected files: {self}")
+            return []
+
         Index = self._Index
 
-        kwargs = {}
-        if doc_ids:
-            with Session(engine) as session:
-                stmt = select(Index).where(
-                    Index.relation_type == "vector",
-                    Index.source_id.in_(doc_ids),  # type: ignore
-                )
-                results = session.execute(stmt)
-                vs_ids = [r[0].target_id for r in results.all()]
-
-            kwargs["filters"] = MetadataFilters(
-                filters=[
-                    MetadataFilter(
-                        key="doc_id",
-                        value=vs_id,
-                        operator=FilterOperator.EQ,
-                    )
-                    for vs_id in vs_ids
-                ],
-                condition=FilterCondition.OR,
+        vrkwargs = {}
```

Comment:
Could you rename this variable to something more meaningful ? It's not obvious to me what `vr` stands for.